lib/doorkeeper/config.rb

```diff
@@ -7,54 +7,38 @@ def initialize
 
   def self.configure(&block)
     @config = Config::Builder.new(&block).build
-    enable_orm
-    check_for_missing_columns
+    setup_orm_adapter
+    setup_orm_models
     setup_application_owner if @config.enable_application_owner?
+    check_requirements
   end
 
   def self.configuration
     @config || (fail MissingConfiguration.new)
   end
 
-  def self.check_for_missing_columns
-    if Doorkeeper.configuration.orm == :active_record &&
-        ActiveRecord::Base.connected? &&
-        ActiveRecord::Base.connection.table_exists?(
-          Doorkeeper::Application.table_name
-        ) &&
-        !Doorkeeper::Application.new.attributes.include?("scopes")
-
-      puts <<-MSG.squish
-[doorkeeper] Missing column: `oauth_applications.scopes`.
-If you are using ActiveRecord run `rails generate doorkeeper:application_scopes
-&& rake db:migrate` to add it.
-      MSG
-    end
+  def self.check_requirements
+    @orm_adapter.check_requirements!(@config)
```

Comment:
I'v no a strong reason, but option like `enable_application_owner`(there just one for now) may need a extra check, so I allow pass `@config` to easy to catch option